contractcourt/contract_resolver.go

```diff
@@ -99,6 +99,13 @@ type ResolverConfig struct {
 	// which contains information about the outcome and should be written
 	// to disk if non-nil.
 	Checkpoint func(ContractResolver, ...*channeldb.ResolverReport) error
+
+	// FetchCanceledHTLCs fetches all canceled HTLCs from the database so
+	// no HTLCs are failed back more than once.
+	FetchCanceledHTLCs func() (fn.Set[uint64], error)
```

Comment:
This should instead be put in `ChannelArbitratorConfig`, similar to `PutResolverReport`

Reply:
wondering why `Checkpoint` is done the other way, but sure will move.

Reply:
Think it's done this way so the resolvers don't need to be exposed to these functions and can access them via the `cfg`.